agent/logging.go

```diff
@@ -167,3 +171,36 @@ func (a *Agent) prepSingleLogger(tc *taskContext, in model.LogOpts, logDir, file
 		Filepath:        filepath.Join(logDir, fileName),
 	}
 }
+
+func redactList(projectVars map[string]string, redacted map[string]bool) []string {
+	suspiciousPatterns := []string{
+		"key",
+		"secret",
+		"auth",
+		"token",
+		"private",
+		"pass",
+		"pw",
+	}
+
+	var redactedKeys []string
+	for key := range projectVars {
+		if _, ok := redacted[key]; ok {
+			// Skip since the key is already in the redacted list.
+			continue
```

Comment:
could it work to not `continue` here and to omit L200-202?

Reply:
There is no guarantee that a redacted project variable will have a suspicious pattern so we definitely want the full list of those and then any additional project vars that are not private but have a suspicious pattern.